### PR TITLE
fixing the variables issue

### PR DESCRIPTION
**Pull request by JAGANNATH-OPTIT**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 1]
```diff
@@ -7,11 +7,11 @@ config_path: "/opt/sparkjobs/ml-analytics-service"
 SYSTEMDDIR: "/lib/systemd/system"
 ml_analytics_version: "release-4.1.0"
 ml_survey_cloud_storage: "AZURE"
-ml_analytics_survey_app_name: "{{ ml_cloud_config.CLOUD_STORAGE }}"
+ml_analytics_survey_app_name: "sunbirdsurvey"
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/defaults/main.yml:10]: make it as a variable and give default value as sunbirdapp.
`"{{ ml_analytics_survey_app_name|default('sunbirdsurvey') }}"`

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 2]
```diff
@@ -7,11 +7,11 @@ config_path: "/opt/sparkjobs/ml-analytics-service"
...
 ml_analytics_integrated_app_name: "sunbird"
 ml_analytics_survey_service: "http://{{private_ingressgateway_ip}}/ml-survey-service"
 ml_analytics_evidence_base_url: "https://{{ml_analytics_azure_account_name}}.blob.core.windows.net/{{ml_analytics_azure_container_name}}/"
-ml_analytics_mongodb_url: "{{ml_analytics_mongodb_url | default(groups['mongo_master'][0]+':27017')}}"
+ml_analytics_mongodb_url: "{{ml_analytics_mongodb | default(groups['mongo_master'][0]+':27017')}}"
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/defaults/main.yml:14]: give the `_url` also. The variables should be self explanatory. and you don't need the `+` sign

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 3]
```diff
@@ -27,7 +27,7 @@ ml_analytics_mongo_projects_collection: "projects"
 ml_analytics_mongo_survey_submissions_collection: "surveySubmissions"
 ml_analytics_mongo_survey_collection: "survey"
 ml_analytics_druid_url:  "{{groups['druid'][0]}}:8081"
-ml_analytics_kafka_url: "{{ml_analytics_kafka_url | default(groups['processing-cluster-kafka'][0]+':9092')}}"
+ml_analytics_kafka_url: "{{ml_analytics_kafka | default(groups['processing-cluster-kafka'][0]+':9092')}}"
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/defaults/main.yml:30]: same thing. `_url` and remove the +.
After these changes, try to template and make sure, the result is correct.

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 4]
```diff
@@ -42,7 +42,7 @@ ml_analytics_azure_account_name: "{{ml_analytics_azure_account_name}}"
 ml_analytics_azure_container_name: "{{ml_analytics_azure_container_name}}"
 ml_analytics_observation_azure_blob_path: "observation/status/"
 ml_analytics_project_azure_blob_path: "reports/"
-ml_analytics_redis_host: "{{groups['ml-redis'][]}}"
+ml_analytics_redis_host: "{{ml_redis_url | default(groups['redis-metadata2'][0])}}"
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/defaults/main.yml:45]: what's this metadata instance? and why the name of `metadata2`

**Review by rjshrjndrn — CHANGES_REQUESTED** [review]

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 5]
```diff
@@ -42,7 +42,7 @@ ml_analytics_azure_account_name: "{{ml_analytics_azure_account_name}}"
 ml_analytics_azure_container_name: "{{ml_analytics_azure_container_name}}"
 ml_analytics_observation_azure_blob_path: "observation/status/"
 ml_analytics_project_azure_blob_path: "reports/"
-ml_analytics_redis_host: "{{groups['ml-redis'][]}}"
+ml_analytics_redis_host: "{{ml_redis_uel | default(groups['dp-redis'][0])}}"
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/defaults/main.yml:45]: It's url not uel

**Review by rjshrjndrn — CHANGES_REQUESTED** [review]

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 6]
```diff
@@ -42,7 +42,7 @@ ml_analytics_azure_account_name: "{{ml_analytics_azure_account_name}}"
 ml_analytics_azure_container_name: "{{ml_analytics_azure_container_name}}"
 ml_analytics_observation_azure_blob_path: "observation/status/"
 ml_analytics_project_azure_blob_path: "reports/"
-ml_analytics_redis_host: "{{groups['ml-redis'][]}}"
+ml_analytics_redis_host: "{{ml_redis_url | default(groups['dp-redis'][0])}}"
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/defaults/main.yml:45]: its not redis url right? it's `ml_redis_host`

Comment (rjshrjndrn) [reply ansible/roles/ml-analytics-service/defaults/main.yml:45]: url is something which has protocol://endpoint. It's just ip address of the service.

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 7]
```diff
@@ -27,7 +27,7 @@ ml_analytics_mongo_projects_collection: "projects"
 ml_analytics_mongo_survey_submissions_collection: "surveySubmissions"
 ml_analytics_mongo_survey_collection: "survey"
 ml_analytics_druid_url:  "{{groups['druid'][0]}}:8081"
-ml_analytics_kafka_url: "{{ml_analytics_kafka_url | default(groups['processing-cluster-kafka'][0]+':9092')}}"
+ml_analytics_kafka_url: "{{ml_kafka_url | default(groups['processing-cluster-kafka'][0:9092)}}"
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/defaults/main.yml:30]: change this to _host too

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 8]
```diff
@@ -7,11 +7,11 @@ config_path: "/opt/sparkjobs/ml-analytics-service"
...
 ml_analytics_version: "release-4.1.0"
 ml_survey_cloud_storage: "AZURE"
-ml_analytics_survey_app_name: "{{ ml_cloud_config.CLOUD_STORAGE }}"
-ml_analytics_integrated_app_name: "sunbird"
+ml_analytics_survey_app_name: "{{ ml_analytics_survey_app_name | default('sunbirdsurvey') }}"
+ml_analytics_integrated_app_name: "{{ ml_analytics_integrated_app_name | default('sunbird') }}"
 ml_analytics_survey_service: "http://{{private_ingressgateway_ip}}/ml-survey-service"
 ml_analytics_evidence_base_url: "https://{{ml_analytics_azure_account_name}}.blob.core.windows.net/{{ml_analytics_azure_container_name}}/"
-ml_analytics_mongodb_url: "{{ml_analytics_mongodb_url | default(groups['mongo_master'][0]+':27017')}}"
+ml_analytics_mongodb_url: "{{ml_mongodb_url | default(groups['mongo_master'][0]:27017)}}"
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/defaults/main.yml:14]: change this to _host too

**Comment by rjshrjndrn** [comment]
Apart from the comments LGTM

File: ansible/roles/ml-analytics-service/defaults/main.yml (hunk 6 above)
Comment (JAGANNATH-OPTIT) [reply ansible/roles/ml-analytics-service/defaults/main.yml:45]: Ok got it. Changing now.

**Review by rjshrjndrn — APPROVED** [review]

**Pull request merged**